components/style/properties/longhand/font.mako.rs

```diff
@@ -361,3 +361,105 @@ ${helpers.single_keyword("font-variant-position",
+        }
+
+        impl ToCss for T {
+            fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+                match *self {
+                    T::Normal => dest.write_str("normal"),
+                    T::Computed(ref ftvs) => {
+                        let mut iter = ftvs.iter();
+                        // handle head element
+                        try!(iter.next().unwrap().to_css(dest));
+                        // handle tail, precede each with a delimiter
+                        for ftv in iter {
+                            try!(dest.write_str(", "));
+                            try!(ftv.to_css(dest));
+                        }
+                        Ok(())
+                    }
+                }
+            }
+        }
+
+        impl ToCss for FeatureTagValue {
+            fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+                let s = format!("\"{}\" {}", self.tag, self.value);
+                dest.write_str(&s)
+            }
+        }
+
+        impl FeatureTagValue {
+            /// <string> [ on | off | <integer> ]
+            pub fn parse(input: &mut Parser) -> Result<FeatureTagValue, ()> {
+                let tag = try!(input.expect_string()).into_owned();
+
+                // allowed strings of length 4 containing chars: <U+20, U+7E>
+                if tag.len() != 4 ||
+                   tag.chars().any(|c| c < ' ' || c > '~') {
+                       return Err(())
+                   }
+
+                if let Ok(value) = input.try(|input| input.expect_integer()) {
+                    // handle integer, throw if it is negative
+                    if value >= 0 {
+                        Ok(FeatureTagValue{ tag: tag, value: value })
+                    } else {
+                        Err(())
+                    }
+                } else if let Ok(_) = input.try(|input| input.expect_ident_matching("on")) {
+                    // on is an alias for '1'
+                    Ok(FeatureTagValue{ tag: tag, value: 1 })
+                } else if let Ok(_) = input.try(|input| input.expect_ident_matching("off")) {
+                    // off is an alias for '0'
+                    Ok(FeatureTagValue{ tag: tag, value: 0 })
+                } else {
+                    // empty value is an alias for '1'
+                    Ok(FeatureTagValue{ tag:tag, value: 1 })
+                }
+            }
+        }
+    }
+
+    #[inline]
+    pub fn get_initial_value() -> computed_value::T {
+        computed_value::T::Normal
+    }
+
+    /// normal | <feature-tag-value>#
+    pub fn parse(_context: &ParserContext, input: &mut Parser) -> Result<SpecifiedValue, ()> {
+        if input.try(|input| input.expect_ident_matching("normal")).is_ok() {
+            Ok(computed_value::T::Normal)
+        } else {
+            input.parse_comma_separated(computed_value::FeatureTagValue::parse)
+                 .map(computed_value::T::Computed)
```

Comment:
Actually, it turns out it already returns an error when passed an empty string (or so it seems, I've added a test for that). I guess it is because the `parse_one` closure is called even when the string is empty, and the `Tag.to_css` throws, because it can't find the required 4-char string.
Should I add a Vec length check anyway? I gotta admit, the current behaviour is far from obvious. Maybe a comment will do?